app/components/Form.js

```diff
@@ -0,0 +1,237 @@
+import React from 'react';
+import { StyleSheet, Text, View, TextInput, TouchableOpacity, AsyncStorage} from 'react-native';
+import { StackNavigator } from 'react-navigation';
+
+export default class Form extends React.Component {
+
+    constructor(props){
+        super(props);
+        this.state={
+            name: '',
+            email: '',
+            emailValidation : true,
+            password: '',
+            passwordValidate: true,
+            showError: false,
+            user: '',
+            errorMsg: '',
+        }
+        this.verifyUser = this.verifyUser.bind(this);
+        this.saveUser = this.saveUser.bind(this);
+        this.showErrorMessage = this.showErrorMessage.bind(this);
+        this.saveLogState = this.saveLogState.bind(this);
+    }
+
+    validate(text,type){
+        var email= /^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i;
+        var num =/^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.{8,})/;
+        if(type=="email"){
+            if (email.test(text)) {
```

Comment:
you can do `this.setState({ emailValidation: email.test(text), email: email.test(text) ? test : ''})` instead of if else